docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -0,0 +1,313 @@
+  }
+}
+--------------------------------------------------
+// TEST[setup:my_index]
+
+
+[discrete]
+[[fix-an-oversharded-cluster]]
+=== Fix an oversharded cluster
+
+If your cluster is experiencing performance issues due to oversharded indices,
+you can use one or more of the following methods to fix them.
+
+[discrete]
+[[reindex-indices-from-shorter-periods-into-longer-periods]]
+==== Create time-based indices that cover longer periods
+
+For time series data, you can create indices that cover longer time intervals.
+For example, instead of daily indices, you can create indices on a monthly or
+yearly basis.
+
+If you're using {ilm-init}, you can do this by increasing the `max_age`
+threshold for the <<ilm-rollover,rollover action>>.
+
+[discrete]
+[[delete-empty-indices]]
+==== Delete empty or unneeded indices
+
+If you're using {ilm-init} and roll over indices based on a `max_age` threshold,
+you can inadvertently create indices with no documents. These empty indices
+provide no benefit but still consume resources.
+
+You can find these empty indices using the <<cat-count,cat count API>>.
+
+[source,console]
+----
+GET /_cat/count/my-index-000001?v
+----
+// TEST[setup:my_index]
+
+Once you have a list of empty indices, you can delete them using the
+<<indices-delete-index,delete index API>>. You can also delete any other
+unneeded indices.
+
+[source,console]
+----
+DELETE /my-index-*
+----
+// TEST[setup:my_index]
+
+[discrete]
+[[force-merge-during-off-peak-hours]]
+==== Force merge during off-peak hours
+
+If you no longer write to an index, you can use the <<indices-forcemerge,force
+merge API>> to <<index-modules-merge,merge>> smaller segments into larger ones.
+This can reduce shard overhead and improve search speeds. However, force merges
+are resource-intensive. If possible, run the force merge during off-peak hours.
+
+[source,console]
+----
+POST /my-index-000001/_forcemerge
+----
+// TEST[setup:my_index]
+
+[discrete]
+[[shrink-existing-index-to-fewer-shards]]
+==== Shrink an existing index to fewer shards
+
+If you no longer write to an index, you can use the
+<<indices-shrink-index,shrink index API>> to reduce its shard count.
+
+[source,console]
+----
+POST /my-index-000001/_shrink/my-shrunken-index-000001
+----
+// TEST[s/^/PUT my-index-000001\n{"settings":{"index.number_of_shards":2,"blocks.write":true}}\n/]
+
+{ilm-init} also has a <<ilm-shrink-action,shrink action>> for indices in the
+warm phase.
+
+[discrete]
+[[reindex-an-existing-index-to-fewer-shards]]
+==== Reindex to an index with fewer shards
+
+You can use the <<docs-reindex,reindex API>> to reindex data from an oversharded
+index to an index with fewer shards. After the reindex, delete the oversharded
+index.
+
+[source,console]
+----
+POST /_reindex
+{
+  "source": {
+    "index": "my-oversharded-index-000001"
+  },
+  "dest": {
+    "index": "my-new-index-000001"
+  }
+}
+----
+// TEST[s/^/PUT my-oversharded-index-000001\n/]
```

Comment:
This is a slower, more resource-intensive version of shrinking, so I wonder if we should mention it at all.

Reply:
+1. Removed with ce5f86c4a9f483461c39672bf03a2e26385a8036.